spyder/api/plugins.py

```diff
@@ -533,3 +557,874 @@ def set_ancestor(self, ancestor):
+        return actions
+
+    def get_action(self, name):
+        """
+        TODO:
+        """
+        actions = self.get_widget().get_actions()
+        if name in actions:
+            return actions[name]
+        else:
+            raise Exception('Action "{0}" not found! Available actions are: '
+                            '{1}'.format(name, actions))
+
+    # --- API: available methods ---------------------------------------------
+    # ------------------------------------------------------------------------
+    def update_title(self):
+        """
+        Update plugin title, i.e. dockwidget or window title.
+        """
+        self.get_widget().update_title()
+
+    def update_margins(self, margin):
+        """
+        Update margins of main widget inside dockable plugin.
+        """
+        self.get_widget().update_margins(margin)
+
+    def switch_to_plugin(self, force_focus=None):
+        """
+        TODO:
+        """
+        self.sig_switch_to_plugin_requested.emit(self, force_focus)
+
+    # --- CoreUI Plugin: This methods need to go in the coreui plugin so that
+    #     application menu/toolbar/status bar handling is managed by that
+    #     plugin
+    # ------------------------------------------------------------------------
+    def create_application_menu(self, name):
+        """
+        Create a Spyder application menu with `name` and return it.
+        """
+        app_menus = getattr(self, '_application_menus', None)
+        if app_menus is None:
+            self._application_menus = {}
+
+        menu = ApplicationMenu(self._main)
+        self._application_menus[name] = menu
+
+        return menu
+
+    def get_application_menu(self, name):
+        """
+        Return an application menu by name.
+        """
+        # TODO: Temporal solution while API for managing app menus is created
+        self._main.source_menu.name = 'source'
+        app_menus = {
+            'source': self._main.source_menu,
+        }
+
+        if name in app_menus:
+            return app_menus[name]
+        else:
+            raise Exception('Application menu "{0}" not found! Available '
+                            'menus are: {1}'.format(name,
+                                                    list(app_menus.keys())))
+
+    def get_application_toolbar(self, name):
+        """
+        Return an application toolbar by name.
+        """
+        # TODO: Temporal solution while API for managing app menus is created
+        app_toolbars = {
+            'source': self.main.source_toolbar,
+        }
+        if name in app_toolbars:
+            return app_toolbars[name]
+        else:
+            raise Exception('Application toolbar "{0}" not found! Available '
+                            'toolbars are: {1}'.format(
+                                name,
+                                list(app_toolbars.keys())))
+
+    def add_item_to_application_menu(self, item, menu, section=None,
```

Comment:
This should be possible with an SpdyerPlugin, right? I mean, a plugin could add an entry to a toolbar or a menu without having a graphical counterpart (as I think we discussed before).

Reply:
Agreed, will move!